File: dehaze-python/app/models/entity/sys_ai_mcp_call.py
from typing import Any

from sqlalchemy import BigInteger, Integer, SmallInteger, String, Text
from sqlalchemy.dialects.mysql import JSON
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import AppendOnlyModel


class SysAiMcpCall(AppendOnlyModel):
    """外部MCP工具调用审计表(只追加，不使用逻辑删除)"""

    __tablename__ = "sys_ai_mcp_call"
    __table_args__ = {"comment": "外部MCP工具调用审计表(只追加)"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    user_id: Mapped[int | None] = mapped_column(
        BigInteger, nullable=True, comment="调用用户ID(关联sys_user.id,NULL表示系统调用)"
    )
    server_id: Mapped[int] = mapped_column(
        BigInteger, nullable=False, comment="关联Server ID(关联sys_ai_mcp_server.id)"
    )
    server_name: Mapped[str | None] = mapped_column(
        String(128), nullable=True, comment="Server名称(冗余快照)"
    )
    tool_name: Mapped[str] = mapped_column(
        String(128), nullable=False, comment="被调用的工具名"
    )
    request: Mapped[Any | None] = mapped_column(
        JSON, nullable=True, comment="调用载荷(JSON)"
    )
    response: Mapped[str | None] = mapped_column(
        Text, nullable=True, comment="响应结果(JSON文本)"
    )
    status: Mapped[int] = mapped_column(
        SmallInteger, nullable=False, default=0, comment="调用状态(0:失败;1:成功)"
    )
    result: Mapped[str] = mapped_column(
        String(16), nullable=False, default="success", comment="调用结果(success;failure)"
    )
    latency_ms: Mapped[int | None] = mapped_column(
        Integer, nullable=True, comment="调用耗时(毫秒)"
    )

File: dehaze-go/internal/model/sys_ai_mcp_call.go
package model

import (
	"encoding/json"
	"time"
)

// SysAiMcpCall 外部 MCP 工具调用审计（只追加，不逻辑删除）
type SysAiMcpCall struct {
	ID         int64           `gorm:"primaryKey;autoIncrement;column:id" json:"id"`
	UserID     *int64          `gorm:"column:user_id;type:bigint;comment:调用用户ID" json:"userId"`
	ServerID   int64           `gorm:"column:server_id;type:bigint;not null;comment:关联Server ID" json:"serverId"`
	ServerName *string         `gorm:"column:server_name;type:varchar(128);comment:Server名称(冗余快照)" json:"serverName"`
	ToolName   string          `gorm:"column:tool_name;type:varchar(128);not null;comment:被调用工具名" json:"toolName"`
	Request    json.RawMessage `gorm:"column:request;type:json;comment:调用载荷" json:"request"`
	Response   *string         `gorm:"column:response;type:text;comment:响应结果" json:"response"`
	Status     int8            `gorm:"column:status;type:tinyint;not null;default:0;comment:调用状态(0:失败;1:成功)" json:"status"`
	Result     string          `gorm:"column:result;type:varchar(16);not null;default:success;comment:调用结果(success/failure)" json:"result"`
	LatencyMs  *int            `gorm:"column:latency_ms;type:int;comment:调用耗时(毫秒)" json:"latencyMs"`
	CreateTime time.Time       `gorm:"column:create_time;type:datetime;not null;autoCreateTime" json:"createTime"`
}

func (SysAiMcpCall) TableName() string {
	return "sys_ai_mcp_call"
}
